derive-input/src/internal/dsl/method.rs

```diff
@@ -1053,6 +1067,20 @@ pub(in crate::internal) fn for_method(
                         false => index_name,
                     };
 
+                    let hook_call_before_update = if spacetimedsl_table.hook_on_update {
+                        let hook_fn_name = format_ident!("on_{}_update", singular_table_name);
+                        quote! {
+                            let old_row = self.ctx().db().#singular_table_name().#primary_key_column_name()
+                                .find(#singular_table_name.#primary_key_column_name)
+                                .expect("Row should exist for update");
```

Comment:
should reuse let_field_name_for_found_value if multi column index checks or reference integrity checks have already found the row. Only if both doesn't have set this to Some(T) instead of the initial None, this code needs to fetch the row.